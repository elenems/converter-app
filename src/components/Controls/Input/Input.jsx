import React from "react";
import TextField from "@material-ui/core/TextField";
export default React.memo(function Input(props) {
  const { isDisabled, inputId, value, fieldLabel, handleChange } = props;
  return (
    <TextField
      disabled={isDisabled}
      id={inputId}
      value={value}
      label={fieldLabel}
      onChange={handleChange}
      variant="outlined"
    />
  );
});
