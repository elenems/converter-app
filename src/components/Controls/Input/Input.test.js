import React from 'react';
import { shallow } from 'enzyme';
import Input from './Input';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });
describe('Input', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<Input debug />);
    expect(component).toMatchSnapshot();
  });
});