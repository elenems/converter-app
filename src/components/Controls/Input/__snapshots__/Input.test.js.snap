// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should render correctly in "debug" mode 1`] = `"<WithStyles(ForwardRef(TextField)) disabled={false} id={[undefined]} value={[undefined]} label={[undefined]} onChange={[undefined]} variant=\\"outlined\\" />"`;
